### ISSUE TITLE
Adding a TabulatedFunction to a CustomIntegrator gives a seg fault

### ISSUE THREAD

**Issue by sgill2** [pandegroup/openmm#2020]
In OpenMM 7.2.1 after adding a `TabulatedFunction` to a `CustomIntegrator` I'm getting a seg fault. Here's some minimal example code that highlights this:
```python
from simtk.openmm.openmm import CustomIntegrator, Discrete1DFunction
tab_values = [0, 5, 20, 25]
tab_funct = Discrete1DFunction(tab_values)
integrator = CustomIntegrator(0.002)
integrator.addTabulatedFunction('test', tab_funct)
print('done')
```
One interesting note about the above example is that the seg fault appears after `print()` successfully prints.

**Comment by peastman** [comment]
That code works fine for me.  This sounds like you might have some sort of incompatibility between libraries.  A few questions:

- What OS are you using?
- Did you install OpenMM with conda or build from source?
- What python are you using?
- Is this the only thing you've seen fail?  For example, do the example scripts like simulatePdb.py work correctly?

**Comment by sgill2** [comment]
I tried this on my computer with MacOS and on a cluster with CentOS linux, both with python 2.7 and 3.5 and got the above behavior. I installed OpenMM from conda. For debugging, I also created a new python 3.5 conda environment on MacOS/Linux with only OpenMM and it's dependencies and had the same behavior occur running the above script. 
I've used OpenMM to run simulations in the past (albeit without using tabulated functions) and they've worked fine for me. Also as a sanity check I just tried `simulatePdb.py` and it ran without issue.

**Comment by peastman** [comment]
Ok, that does sound like a real problem, so I'll look into it.  One thing that comes to mind is that it might be an error in the SWIG wrapper, such that it's getting confused about which Python object owns which C++ object.  That could lead to it accessing memory that has already been freed, or perhaps trying to free the same memory twice.  Anyway, that's the first thing I'll check for.

**Comment by peastman** [comment]
That was exactly the problem.  I just sent a PR with the fix: #2023.

**Comment by sgill2** [comment]
That's great! Thanks for the help.
